http/src/main/scala/org/http4s/blaze/http/http2/Http2Connection.scala

```diff
@@ -47,14 +47,17 @@ trait Http2ClientConnection extends Http2Connection with Http2ClientSession {
 }
 
 object Http2Connection {
-  sealed trait ConnectionState
+  sealed abstract class ConnectionState {
```

Comment:
This is probably true a lot of places, but do you want to extend Product with Serializable for better inference?

Reply:
Is this just to avoid the inference of the common subtype `ConnectionState with Product with Serializable` since all the sub types are case *things*? 
(this is the first time I've heard of this strategy)

But yeah, I can do that if you like.

Reply:
Yep.  I've started habitually making all the types this to improve the inference.